components/Marketing/Trial.js

```diff
@@ -0,0 +1,211 @@
+    setEmail({
+      ...email,
+      value,
+      error: (
+        (value.trim().length <= 0 && t('Marketing/Trial/email/error/empty')) ||
+        (!isEmail(value) && t('Marketing/Trial/email/error/invalid'))
+      ),
+      dirty: shouldValidate
+    })
+  }
+
+  const requestAccess = e => {
+    e.preventDefault && e.preventDefault()
+
+    setLoading(true)
+    setServerError()
+
+    if (!me) {
+      handleEmail(email.value, true)
+
+      if (errorMessages.length > 0) {
+        setLoading(false)
+        return setShowErrors(true)
+      }
+
+      return props.signIn(
+        email.value,
+        'trial',
+        consents,
+        tokenType
+      )
+        .then(({ data: { signIn } }) => {
+          setTokenType(signIn.tokenType)
+          setPhrase(signIn.phrase)
+
+          setLoading(false)
+          setSigningIn(true)
+        })
+        .catch(error => {
+          setServerError(error)
+          setLoading(false)
+        })
+    }
+
+    setSigningIn(false)
+
+    props.requestAccess()
+      .then(() => toOnboarding())
+      .catch(error => {
+        setServerError(error)
+        setLoading(false)
+      })
+  }
+
+  const reset = () => {
+    setLoading(false)
+    setSigningIn(false)
+  }
+
+  const errorMessages = [email.error]
+    .concat(getConsentsError(t, REQUIRED_CONSENTS, consents))
+    .filter(Boolean)
+
+  return (
+    <Fragment>
+      <H1>Danke für Ihre Neugier!</H1>
+      <P style={{ marginTop: 40 }}>
+        Bitte tragen Sie unten Ihre E-Mail-Adresse ein. Im Gegenzug laden wir Sie auf einen Streifzug
+        durch die Republik ein. Gerne schicken wir Ihnen zudem einen aktuellen Newsletter, dann sehen
+        Sie einen Tag lang, worüber wir berichten.
+      </P>
+
+      <form onSubmit={requestAccess}>
+        {!me && (
+          <div style={{ opacity: (signingIn) ? 0.6 : 1, marginTop: 20 }}>
+            <Field
+              label={t('Marketing/Trial/email/label')}
+              value={email.value}
+              error={email.dirty && email.error}
+              dirty={email.dirty}
+              disabled={signingIn}
+              onChange={(_, value, shouldValidate) => handleEmail(value, shouldValidate)} />
+            <div style={{ marginTop: 40 }}>
+              <Consents
+                required={REQUIRED_CONSENTS}
+                accepted={consents}
+                disabled={signingIn}
+                onChange={setConsents} />
+            </div>
+          </div>
+        )}
+
+        {showErrors && errorMessages.length > 0 && (
+          <div {...styles.errorMessages}>
+            {t('Marketing/Trial/error/title')}<br />
+            <ul>
+              {errorMessages.map((error, i) => (
+                <li key={i}>{error}</li>
+              ))}
+            </ul>
+          </div>
+        )}
+
+        {!signingIn && (
+          <div {...styles.button}>
+            {loading
+              ? <InlineSpinner />
+              : <Button
```

Comment:
The button could be disabled until the form is valid.

Reply:
Having it not disabled initially allows for a user to click it and see a list of errors what requires to be filled.

I like this pattern quite much and we're using it in other forms, too. Would like to keep it that way.